### PR TITLE
Refactored notifications & extensions.

### PR DESCRIPTION
**Pull request by 0xWDG**
## Summary/Changes Made

- Refactored notifications to support passing the sender name,
- Added support for JSON based views using 'DynamicUI',
- Moved the broadcaster handler to its own function/extension.
- Created 'ExtensionDoc_fromAE' to document actions which can be received by an extension
- Created 'ExtensionDoc_toAE' to document which actions can be sent by an extension"

## TODO

None

## Motivation

Extension system

## Testing

Nope.

## Screenshots/GIFs

image

_How to replicate_:
create `~/Library/Application Support/com.auroraeditor/Extensions/test.JSext/extension.js`
with those contents:
```javascript
AuroraEditor.respond("openWindow", {
  view: '[{"type": "Text","title": "Wait, am i generating views from JSON?"}]',
});
```

## Checklist

Please ensure all of the following are completed before submitting the PR:

- [x] Code has been tested and verified.
- [x] Documentation has been updated to reflect the changes.
- [x] All tests pass successfully.
- [x] Code follows the project's coding guidelines and style.
- [x] The branch is up-to-date with the latest changes from the main branch.
- [x] Reviewed by at least one other contributor.

## Related Issues

This PR addresses the following issue(s): None.

## Additional Notes

None.

### CODE REVIEW

**Comment by aurora-care-bear** [comment]
Thanks for submitting a pull request!
We will review it as soon as possible.

File: AuroraEditor/Core/ExtensionSystem/ExtensionDocumentation/ExtensionDoc_fromAE.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    /// Did dismiss an notification
+    ///
+    /// - Parameter identifier: Notification identifier
+    /// - Parameter extension: Extension name
+    func didDismissNotification(identifier: String, extension: String)
+
+    /// No operation.
+    func noop()
+}
```
Comment (nanashili) [inline AuroraEditor/Core/ExtensionSystem/ExtensionDocumentation/ExtensionDoc_fromAE.swift:104]: Is this just for testing? If so please add a not for testing purposes only

**Review by nanashili — DISMISSED** [review]

File: AuroraEditor/Core/ExtensionSystem/ExtensionDocumentation/ExtensionDoc_fromAE.swift (hunk 1 above)
Comment (0xWDG) [reply AuroraEditor/Core/ExtensionSystem/ExtensionDocumentation/ExtensionDoc_fromAE.swift:104]: It's not for testing. it's the command where the broadcaster init's with.

Comment (0xWDG) [reply AuroraEditor/Core/ExtensionSystem/ExtensionDocumentation/ExtensionDoc_fromAE.swift:104]: Noop is supported, but not documented anymore

**Review by nanashili — APPROVED** [review]

**Comment by aurora-care-bear** [comment]
Hey @0xWDG,

Thanks for your contribution.
Your PR has been merged!
Your name will appear on https://auroraeditor.com/#contributors in a few moments, if you have contributed to the editor.

Thanks again for your contribution!

Kind Regards,
Aurora Editor Team.

**Pull request merged**